tap_sftp/client.py

```diff
@@ -40,19 +40,20 @@ def __try_connect(self):
             self.transport = paramiko.Transport((self.host, self.port))
             self.transport.use_compression(True)
             self.transport.connect(username = self.username, password = self.password, hostkey = None, pkey = self.key)
-        except AuthenticationException as ex:
+            self.sftp = paramiko.SFTPClient.from_transport(self.transport)
+        except (AuthenticationException, SSHException) as ex:
             self.transport.close()
             self.transport = paramiko.Transport((self.host, self.port))
             self.transport.use_compression(True)
             self.transport.connect(username= self.username, password = self.password, hostkey = None, pkey = None)
+            self.sftp = paramiko.SFTPClient.from_transport(self.transport)
 
     def __ensure_connection(self):
         if not self.__active_connection:
             try:
                 self.__try_connect()
-            except AuthenticationException as ex:
+            except (AuthenticationException, SSHException) as ex:
```

Comment:
Interesting, I didn't realize you can catch a raw exception thrown in the handling of a lower exception handler. But :themoreyouknow:
```
>>> try:
...     try:
...         raise ValueError
...     except ValueError as e:
...         raise Exception("Caught?")
... except Exception as ex:
...     raise Exception("CAUGHT!!!!") from ex
...
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
ValueError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
Exception: Caught?

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 7, in <module>
Exception: CAUGHT!!!!
```

Reply:
Hmmmm its probably best to not do that even if it does work. Restructured the code a little.